### PR TITLE
Nukkit support

### PR DESCRIPTION
**Pull request by DevSrSouza**

### CODE REVIEW

**Review by stephan-gh — CHANGES_REQUESTED** [review]
Please run `./gradlew licenseFormat` to add the necessary license headers and commit the result. Thanks :)

**Comment by mikroskeem** [comment]
Don't forget to squash commits as well =)

**Comment by DevSrSouza** [comment]
Done :333

**Review by mikroskeem — APPROVED** [review]

File: src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        if (description.version == null) {
+            throw InvalidPluginDescriptionException("Plugin version class is not set")
+        }
+
+        if (description.api == null || description.api!!.isEmpty()) {
+            throw InvalidPluginDescriptionException("Nukkit API version is not set")
+        }
+    }
+}
```
Comment (stephan-gh) [inline src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt]: Can you add a new line here?

File: src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        if (description.main == null) {
+            throw InvalidPluginDescriptionException("Main class is not defined")
+        }
+
+        if (description.version == null) {
+            throw InvalidPluginDescriptionException("Plugin version class is not set")
+        }
+
+        if (description.api == null || description.api!!.isEmpty()) {
```
Comment (stephan-gh) [inline src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt:53]: This looks like `description.api.isNullOrEmpty()`: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-null-or-empty.html

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt (hunk 1 above)
Comment (DevSrSouza) [reply src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt]: Done :3

**Review by stephan-gh — APPROVED** [review]

File: src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt (hunk 2 above)
Comment (DevSrSouza) [reply src/main/kotlin/net/minecrell/pluginyml/nukkit/NukkitPlugin.kt:53]: description.api is a list :(
https://github.com/NukkitX/Nukkit/blob/master/src/main/java/cn/nukkit/plugin/PluginDescription.java#L114

**Pull request merged**